src/components/FormTransaction/FormTransaction.jsx

```diff
@@ -5,18 +5,43 @@ import css from "./FormTransaction.module.css";
 import Button from "../Button/Button.jsx";
 import PropTypes from "prop-types";
 
-function FormTransaction({ selectOptions }) {
+function FormTransaction({
+  selectOptions,
+  onAddTransaction,
+  onClearTransactions,
+}) {
   const [startDate, setStartDate] = useState(new Date());
   const [description, setDescription] = useState("");
   const [category, setCategory] = useState("");
   const [amount, setAmount] = useState("");
 
-  const handleSubmitInput = () => {
-    // Logika obsługi przycisku INPUT do zrobienie
+  const handleSubmitInput = (e) => {
+    e.preventDefault();
+    if (!description || !category || !amount) {
+      alert("Please fill all the fields.");
+      return;
+    }
+
+    const newTransaction = {
+      id: Date.now(), // BACKEND zakualizować id
+      date: startDate.toISOString().split("T")[0],
+      description,
+      category,
+      sum: parseFloat(amount) * -1, //będzie się wyświetlać ujemna wartość w tabeli
+    };
```

Comment:
tutaj musisz dodać zmienną w zależności czy to jest income czy expense albo zrobić 2 funkcje bo za każdym razem przekazywany jest minus

Reply:
poprawione